src/modules/addimage.js

```diff
@@ -730,20 +734,28 @@ import { atob, btoa } from "../libs/AtobBtoa.js";
   var arrayBufferToBinaryString = (jsPDFAPI.__addimage__.arrayBufferToBinaryString = function(
     buffer
   ) {
-    try {
-      return atob(btoa(String.fromCharCode.apply(null, buffer)));
-    } catch (e) {
-      if (
-        typeof Uint8Array !== "undefined" &&
-        typeof Uint8Array.prototype.reduce !== "undefined"
-      ) {
-        return new Uint8Array(buffer)
-          .reduce(function(data, byte) {
-            return data.push(String.fromCharCode(byte)), data;
-          }, [])
-          .join("");
+    // Skip direct conversion as it might take many hundred milliseconds before throwing.
+    if (buffer.length < ARRAY_BUFFER_LIMIT) {
+      try {
+        return atob(btoa(String.fromCharCode.apply(null, buffer)));
+      } catch (e) {
+        // Buffer was overflown, fall back to other methods
       }
```

Comment:
I don't think this works, the second argument is an `ArrayBuffer`, which cannot be indexed and therefore cannot be used in `.apply`. You need to convert to `Uint8Array` first.

Reply:
Also, I know `atob(btoa(str))` was in the previous code, but I don't think `atob(btoa(str))` is any different from `str`. Is there an example that differs?

Reply:
Was wondering about that as well.